kie-wb-tests/kie-wb-tests-gui/src/test/java/org/kie/wb/selenium/ui/LoadAllPerspectivesIntegrationTest.java

```diff
@@ -45,11 +46,10 @@ public void allPerspectivesCanBeLoaded() {
                 AbstractPerspective perspective = home.getNavbar().navigateTo( _p );
 
                 //ProjectAuthoring will direct to ProjectLibrary if there are no Projects in the workbench
-                if ( perspective instanceof ProjectAuthoringPerspective ) {
-                    if ( !perspective.isDisplayed() ) {
-                        LOG.info( "ProjectAuthoringPerspective not displayed. Trying fallback ProjectLibraryPerspective.." );
-                        _p = Persp.PROJECT_LIBRARY;
-                        perspective = home.getNavbar().navigateTo( _p );
+                if (perspective instanceof ProjectAuthoringPerspective) {
+                    if (((ProjectAuthoringPerspective) perspective).isAuthoringDisabled()) {
```

Comment:
The second method name is "isAuthoring**Disabled**" which serves to determine whether Authoring perspective exists to begin with or the request to this perspective just redirects the user to Library perspective (as no projects are created in/imported to Workbench). The method is unique and tied only to this perspective as only Project Authoring shows this redirection behavior.

"isDisplayed" method, on the other hand, serves to validate whether the content of the Authoring perspective is displayed properly when it does exist. This method is part of the interface which is being implemented by all other perspectives, so if you still think that having two separate methods to distinguish these two scenarios, I can think of alternatives of removing "isAuthoringDisabled" method.
One alternative would be to use "isDisplayed" method of ProjectLibraryPerspective instead of "isAuthoringDisabled", but the logic around the code we are discussing here would be more complex and hence harder to read.